jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -249,7 +248,24 @@ public Authorization authenticatePush() throws RegistryAuthenticationFailedExcep
 
   @VisibleForTesting
   URL getAuthenticationUrl(String scope) throws MalformedURLException {
-    return new URL(authenticationUrlBase + scope);
+    return new URL(realm + "?" + getUrlEncodedServiceScope(scope));
+  }
+
+  @VisibleForTesting
+  String getAuthenticationFormUrlEncoded(String scope, String token) {
```

Comment:
This method name sounds misleading to me, since it doesn't return a plain string before url-encoding. How about `getOAuth2AuthRequestParameters()`?

Reply:
I still don't see a test.

Reply:
see test case in last commit